### PR TITLE
pythonPackages.fpylll: init at 0.3.0dev

### PR DESCRIPTION
**Pull request by timokau**
###### Motivation for this change

~~This depends on #38781, do not merge before that.~~

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/fpylll/default.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    cython
+    cysignals
+  ];
+
+  nativeBuildInputs = [
+    pkgconfig
+  ];
+
+  meta = {
+    description = "A Python interface for fpylll";
```
Comment (dotlambda) [inline pkgs/development/python-modules/fpylll/default.nix]: A Python interface for fplll

**Review by dotlambda — CHANGES_REQUESTED** [review]
The website says

> fpylll also relies on py.test for testing Python

Please specify an appropriate `checkPhase`.

**Comment by timokau** [comment]
I added the `checkPhase`, added a fplll upgrade that is necessary for fpylll and added a numpy dependency to pass the tests.

File: pkgs/development/python-modules/fpylll/default.nix [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  ];
+
+  checkPhase = ''
+    py.test
+  '';
+
+  meta = {
+    description = "A Python interface for fplll";
+    homepage = https://github.com/fplll/fpylll;
+    maintainers = with lib.maintainers; [ timokau ];
```
Comment (dotlambda) [inline pkgs/development/python-modules/fpylll/default.nix]: License?

File: pkgs/development/python-modules/fpylll/default.nix [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  };
+
+  buildInputs = [
+    gmp
+    pari
+    mpfr
+    fplll
+    cython
+    cysignals
+    numpy
```
Comment (dotlambda) [inline pkgs/development/python-modules/fpylll/default.nix]: Some of these belong in `propagatedBuildInputs`.

File: pkgs/development/python-modules/fpylll/default.nix (hunk 2 above)
Comment (timokau) [reply pkgs/development/python-modules/fpylll/default.nix]: Added.

File: pkgs/development/python-modules/fpylll/default.nix [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+  checkPhase = ''
+    py.test
+  '';
+
+  meta = {
+    description = "A Python interface for fplll";
+    homepage = https://github.com/fplll/fpylll;
+    maintainers = with lib.maintainers; [ timokau ];
+    license = lib.licenses.gpl2;
```
Comment (dotlambda) [inline pkgs/development/python-modules/fpylll/default.nix]: > fpylll is licensed under the GPLv2+

https://github.com/fplll/fpylll#attribution--license

**Comment by dotlambda** [comment]
@GrahamcOfBorg build python2.pkgs.fpyll python3.pkgs.fpyll

**Comment by dotlambda** [comment]
@GrahamcOfBorg build python2.pkgs.fpyll python3.pkgs.fpylll

**Comment by dotlambda** [comment]
@GrahamcOfBorg build python2.pkgs.fpylll python3.pkgs.fpylll

**Comment by timokau** [comment]
```
building of '/nix/store/l60jan1mgs2yd0lnkmsprzjapx4rynv9-python3.6-Cython-0.28.1.drv' timed out after 3600 seconds
```

Maybe ofBorg should indicate when a failure is due to a timeout.

**Pull request merged**